language/tools/test_generation/src/control_flow_graph.rs

```diff
@@ -266,20 +266,52 @@ impl CFG {
                 basic_block.locals_in =
                     cfg_copy.merge_locals(cfg_copy.get_parent_ids(block_id as BlockIDSize));
             }
-            // basic_block.locals_out = basic_block.locals_in.clone();
             basic_block.locals_out = CFG::vary_locals(&mut rng, basic_block.locals_in.clone());
         }
     }
 
+    /// Decide the serialization order of the blocks in the CFG
+    pub fn serialize_block_order(&self) -> Vec<BlockIDSize> {
+        let mut block_order: Vec<BlockIDSize> = Vec::new();
+        let mut block_queue: VecDeque<BlockIDSize> = VecDeque::new();
+        block_queue.push_back(0);
+        while !block_queue.is_empty() {
+            let block_id_front = block_queue.pop_front();
```

Comment:
We seem to need a MIRAI model for pop_front, etc.

Reply:
Yes, that's necessary. I have it on my todo list to do this if time permits.